src/cage/src/timer.rs

```diff
@@ -0,0 +1,144 @@
+// Timer functions for Rust interface.
+#![allow(dead_code)]
+
+use std::sync::{Arc, Mutex, MutexGuard};
```

Comment:
I know you didn't explicitly write the code in this file but this is a good chance for us to add at least a blurb on what the interval timer is used for (alarm etc) and why we're not letting the kernel handle this